src/services/productService.js

```diff
@@ -1,13 +1,19 @@
 const productRepo = require('../repositories/productRepo');
+const Product = require('../models/products'); 
+const Review = require('../models/reviews');
 
 const getAllProducts = async () => productRepo.getAllProducts();
 
 const create = async (product) => productRepo.save(product);
 
 const findById = async (id) => productRepo.findByIdWithReviews(id);
 
+const deleteProductAndReviews = async (productId) => productRepo.deleteProductAndReviews(productId);
+
```

Comment:
Why are we not returing promise from here? How are you ensuring the product is being deleted here?